WalletWasabi.Gui/Tabs/WalletManager/WalletManagerViewModel.cs

```diff
@@ -96,102 +98,96 @@ public ViewModelBase CurrentView
 		public override void OnOpen()
 		{
 			base.OnOpen();
-
-			Dispatcher.UIThread.PostLogException(async () =>
+			if (Disposables != null)
 			{
-				await RefreshHardwareWalletListAsync();
-				HardwareWalletRefreshCancel?.Dispose();
-				HardwareWalletRefreshCancel = null;
-			});
+				throw new Exception("Send tab opened before last one closed.");
+			}
+
+			Disposables = new CompositeDisposable();
+
+			Observable.FromEventPattern<IEnumerable<HardwareWalletInfo>>(
+				Global.HwiService,
+				nameof(Global.HwiService.NewEnumeration))
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(e =>
+				{
+					IEnumerable<HardwareWalletInfo> hwis = e.EventArgs;
+					RefreshHardwareWalletListAsync(hwis);
+				}).DisposeWith(Disposables);
+
+			Global.HwiService.Start();
 		}
 
-		private CancellationTokenSource HardwareWalletRefreshCancel { get; set; }
 		private bool HwTabAutomaticallySelectedOnce { get; set; } = false;
 
-		private async Task RefreshHardwareWalletListAsync()
+		private void RefreshHardwareWalletListAsync(IEnumerable<HardwareWalletInfo> hwis)
```

Comment:
If it is not async anymore you could rename it (remove Async from its name).

Reply:
Yes!